text/2-aurgy.md

```diff
@@ -0,0 +1,135 @@
+
+[This article is very interesting]: https://github.com/UCLA-Creative-Labs/cl-rfcs/issues/2
+
+## Why are we doing this?
+
+Spotify's Blends are cool, but they're still limited. Being able to listen to
+things as a group would be a great feature to have. Also, being able to generate
+group playlists based on themes would be great to have for all kinds of events
+and occasions---saves everyone the trouble of passing the phone around and searching
+up songs.
+
+## What is the high level design/implementation?
+
+The best plan of attack for this project would be to create a PWA. Realistically, this
+would be a mobile-first application, but by creating a PWA we can a) support desktop
+and b) avoid working with something like React Native (bleh) but still achieve cross-
+platform support.
+
+This project has a few key components:
+* Creating and managing a group/lobby
+* Generating a playlist
+* Playing music from the playlist
+
+Let's delve into these in more detail.
+
+#### Creating and managing a group/lobby
+
+In this project, users should be able to join a private, protected group. We could probably
+accomplish this using a standard lobby system: one person acts as the host and creates the
+lobby and receives a unique code; users who then enter this code can enter that particular
+lobby.
+
+We could probably set something like this up with a simple document database. Whenever a host
+creates a lobby, a new document is created with a unique identifying code. Then, if another
+user enters that code, their ID is added to the document. The document might look something
+like this:
+```json
+"<unique id>": {
+  "users": [
+    "<id>", "<id>", ...
+  ]
+}
+```
+
+In the interim, something like Firebase Firestore should be enough. It might be a good idea
+to switch to something else (like MongoDB perhaps) if we want to worry about scaling and
+whatnot.
+
+We could achieve persistent groups by just keeping these unique identifiers persistent, i.e.
+not deleting them. This way, the host and any other users can re-enter groups they are a part
+of by just entering their unique code.
+
+#### Generating a playlist
+
+Here's the meat of the project. We would make heavy use of the [Spotify API] and all of the
+information available about individual tracks here.
+
+First, we need a list of themes that the user can choose from. Some suggestions are "road trip",
+"party", or "kickback". Then, we would have to correlate each theme with a set of track features,
+such as genre.
+
+We would also need to understand the users' music preferences. Again, we would pull things like
+favorite (recent?) genres and figure out areas of strongest intersection between users.
+
+Finally, we want to take the features prescribed by the desired theme and the features that are
+most common among users, and pull songs that match as many of these features as possible.
+
+Once the playlist is generated, we can add it to the host's library.
```

Comment:
**EDIT**

On second pass, i think u meant the below, but let's add it to the RFC

---

So the way Spotify does this is they create the playlist on their own account, and then share it to the applicable parties.

I think we should follow a similar model because:
1. we wouldn't have to synchronize different user's libraries
2. update would happen to only one playlist

Reply:
I actually did not mean that, but that's a better model. will add